player/player.gd

```diff
@@ -108,3 +112,13 @@ func _handle_player_hit(amount: int) -> void:
 	
 	if(new_health == 0):
 		SignalBus.player_died.emit()
+
+# These would probably be simpler if we create gun and magazine classes
+func _handle_clip_size_upgrade(level: int):
+	_max_ammo = 7 + level
+	SignalBus.player_ammo_changed.emit(_ammo, _max_ammo)
+
+# This one seems to be bugged a bit not sure
+func _handle_fire_rate_upgrade(level: int):
+	var base = 0.2
```

Comment:
the player's initial fire rate is `0.1` so it makes the fire rate slower after the first upgrade, but it's a separate problem having start/base values defined in different place. just mentioning it.